cranelift/src/souper_harvest.rs

```diff
@@ -77,13 +98,33 @@ pub fn run(options: &Options) -> Result<()> {
 
     let (send, recv) = std::sync::mpsc::channel::<String>();
 
-    let writing_thread = std::thread::spawn(move || -> Result<()> {
-        for lhs in recv {
-            output
-                .write_all(lhs.as_bytes())
-                .context("failed to write to output file")?;
+    let writing_thread = std::thread::spawn({
+        let output_dir = options.output_dir.clone();
+        let keep_harvest_source = options.add_harvest_source;
+        move || -> Result<()> {
+            let mut already_harvested = HashSet::new();
+            for lhs in recv {
+                let lhs = if keep_harvest_source {
+                    &lhs
+                } else {
+                    // Remove the first `;; Harvested from v12 in u:34` line.
+                    let i = lhs.find('\n').unwrap();
+                    &lhs[i + 1..]
+                };
+                let hash = fxhash::hash(lhs.as_bytes());
+                if already_harvested.insert(hash) {
```

Comment:
Keeping the set of hashes might skip some results due to collisions, right? Are the `lhs` strings too big to store in the HashSet instead?

Reply:
Yeah. I just didn't want to keep all the LHSs in memory and blake3 is pretty good an not-yet-known-to-be-broken so it seems safe. Also, I'm pretty sure I'm the only user of this tool :-p